app/components/autocomplete/channel_mention/channel_mention.js

```diff
@@ -82,21 +82,17 @@ export default class ChannelMention extends PureComponent {
         }
 
         if (matchTerm !== this.props.matchTerm) {
-            // if the term changed and we haven't made the request do that first
             const {currentTeamId} = this.props;
             this.runSearch(currentTeamId, matchTerm);
-            return;
         }
 
-        if (requestStatus !== RequestStatus.STARTED &&
-            (myChannels !== this.props.myChannels || otherChannels !== this.props.otherChannels ||
-                privateChannels !== this.props.privateChannels || publicChannels !== this.props.publicChannels ||
-                directAndGroupMessages !== this.props.directAndGroupMessages ||
-                myMembers !== this.props.myMembers || deletedPublicChannels !== this.props.deletedPublicChannels)) {
-            // if the request is complete and the term is not null we show the autocomplete
+        if (matchTerm === '' || (myChannels !== this.props.myChannels || otherChannels !== this.props.otherChannels ||
```

Comment:
Unrelated to this PR, but this conditional needs some refactoring :)

Reply:
Proposal?

Reply:
It might be nice to pull out the `matchTerm === ''` check from the rest of the props-did-change tests, since it seems to be there for different reasons.

Less thought through, but perhaps `sections` could be moved out of state and generated on `render()`, with this component refactored to use `shouldComponentUpdate ` instead? It would leave the conditional intact, but at least focus its meaning better.

Obviously, out-of-scope for this PR :)